File: Intervals/252.cpp
#include <algorithm>
#include <vector>
using namespace std;


class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        if (intervals.empty()) {
            return true;
        }
        
        sort(intervals.begin(), intervals.end());
        for (int i = 0; i < intervals.size() - 1; i++) {
            if (intervals[i][1] > intervals[i + 1][0]) {
                return false;
            }
        }
        return true;
    }
};

File: Intervals/252.rs


impl Solution {
    pub fn can_attend_meetings(intervals: Vec<Vec<i32>>) -> bool {
        
        if intervals.len() == 0 {
            return true;
        }
        
        let mut intervals = intervals;
        intervals.sort();
        
        for i in 0..(intervals.len() - 1) {
            if intervals[i][1] > intervals[i + 1][0] {
                return false;
            }
        }
        return true;
    }
}